### PR TITLE
[tune] Fix _SafeFallbackEncoder type checks

### PR DESCRIPTION
**Pull request by hartikainen**
## What do these changes do?
Currently, the type checks in logger.py result in a bunch of warnings like this:
```
/home/kristian/github/hartikainen/ray/python/ray/tune/logger.py:1: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
```

This fixes those type checks.

## Related issue number
n/a

### CODE REVIEW

**Review by ericl — APPROVED** [review]
LGTM

File: python/ray/tune/logger.py [hunk 1]
```diff
@@ -255,10 +257,18 @@ def default(self, value):
...
+            if (type(value).__module__ == np.__name__
+                    and isinstance(value, np.ndarray)):
+                return value.tolist()
+
+            if issubclass(type(value), numbers.Integral):
                 return int(value)
+            if issubclass(type(value), numbers.Number):
+                return float(value)
+
+            return super(_SafeFallbackEncoder, self).default(value)
```
Comment (ericl) [inline python/ray/tune/logger.py]: This looks ok, but is it necessary?

Comment (hartikainen) [reply python/ray/tune/logger.py]: I don't know if it's necessary but it seemed weird that there could potentially be values that just go past this try block without being properly encoded (i.e. encoded as `None`).

File: python/ray/tune/logger.py [hunk 2]
```diff
@@ -254,11 +256,19 @@ def __init__(self, nan_str="null", **kwargs):
     def default(self, value):
         try:
             if np.isnan(value):
-                return None
-            if np.issubdtype(value, float):
-                return float(value)
-            if np.issubdtype(value, int):
+                return self.nan_str
```
Comment (hartikainen) [inline python/ray/tune/logger.py]: I assume this is supposed to be `self.nan_str` and not `None`. Correct me if I'm wrong.

Comment (hartikainen) [reply python/ray/tune/logger.py]: Actually, I think the result would be the same whether using `None` or `"null"` (which is the default of `self.nan_str`). Is there a reason for using `self.nan_str`?

Comment (richardliaw) [reply python/ray/tune/logger.py]: I guess it would be better, especially if we reused _SafeFallbackEncoder somewhere else

**Comment by richardliaw** [comment]
jenkins retest this please

**Comment by richardliaw** [comment]
jenkins retest this please

**Pull request merged**